### PR TITLE
Fix glitch in undo/redo of note edits via the menu

### PR DESCRIPTION
**Pull request by thmueller64**
Addresses #5722.

### CODE REVIEW

File: src/gui/editors/PianoRoll.cpp [hunk 1]
```diff
@@ -3563,13 +3563,14 @@ void PianoRoll::focusOutEvent( QFocusEvent * )
...
 
-void PianoRoll::focusInEvent( QFocusEvent * )
+void PianoRoll::focusInEvent( QFocusEvent * ev )
 {
 	if ( hasValidPattern() )
 	{
 		// Assign midi device
 		m_pattern->instrumentTrack()->autoAssignMidiDevice(true);
 	}
+    QWidget::focusInEvent(ev);
```
Comment (zonkmachine) [inline src/gui/editors/PianoRoll.cpp:3573]: You have 4 spaces in front of this line. Please change that to 1 tab.

**Comment by zonkmachine** [comment]
I've tested this and it fixes #5722

**Comment by Veratil** [comment]
@zonkmachine can you make sure this doesn't interfere with any other normal operation?

**Comment by zonkmachine** [comment]
> @zonkmachine can you make sure this doesn't interfere with any other normal operation?

Yes, I need to test if the commit that caused this issue still works.

**Comment by superpaik** [comment]
Undo via menu works well. Via hotkey too.
But there is something weird with redo. Via menu works. But via hotkeys there is a strange behaviour. Steps:
- put three notes on the piano roll.
- undo via menu (last note is deleted) OK
- redo via hotkey doesn't work. KO
- undo again via menu (second note is deleted) OK
- redo via hotkey works well (second note appears) redo via hotkey again (last note appears) OK

**Review by DomClark — APPROVED** [review]

**Comment by thmueller64** [comment]
@superpaik : That's strange indeed. I can also reproduce this with the current master, so it appears not to be related to this PR. Maybe file another issue if you can also confirm this on master?

**Comment by superpaik** [comment]
@thmueller64 Yes, it happens in the master as well, so it's not related to this PR.
I won't create another issue because there are few already related to undo/redo and the piano roll, and I've been told that this functionality (undo/redo) needs some kind of refactoring because of it's glitches.
thanks!

**Comment by zonkmachine** [comment]
> > @zonkmachine can you make sure this doesn't interfere with any other normal operation?
> 
> Yes, I need to test if the commit that caused this issue still works.

Some testing. At a quick workthrough everything looks fine to me. I'm not sure I tested everything related to 5d7e6720e1ba0c154041fd663801f7a7a0e16cb5 . I don't have time to do a deeper check right now but I don't think that's necessary. I think this is fine to merge.

**Review by zonkmachine — APPROVED** [review]

**Comment by zonkmachine** [comment]
I could replicate the issue that @superpaik describes but only once. It's definitely intermittent for me. 
Merging!

**Pull request merged**